test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+                .as_str()
+                .expect("not a string")
+                .parse::<u64>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _id_lsb = match value.get("id").and_then(|resource| resource.get("lsb")) {
+            Some(_id_lsb) => _id_lsb
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u64>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let ___id = UUID {
+            msb: _id_msb,
+            lsb: _id_lsb,
+            special_fields: SpecialFields::default(),
+        };
+      
+        let __id = MessageField(Some(Box::new(___id)));
+
+        let _ttl = match value.get("ttl") {
+            Some(_ttl) => _ttl
+                .as_str()
+                .unwrap_or_else(|| panic!("Deserialize: something wrong with ttl field"))
+                .parse::<u32>()
+                .expect("ttl parsing error"),
+            None => 0,
+        };
+
+        let _permission_level = match value.get("permission_level") {
+            Some(_permission_level) => _permission_level
+                .as_str()
+                .unwrap_or_else(|| {
+                    panic!("Deserialize: something wrong with permission_level field")
+                })
+                .parse::<u32>()
+                .expect("permission_level parsing error"),
+            None => 0,
+        };
+
+        let _commstatus = match value.get("commstatus") {
+            Some(_commstatus) => from_str_comstatus(
+                _commstatus
+                    .as_str()
+                    .expect("Deserialize:something wrong with commstatus field"),
+            ),
+            None => UCode::OUT_OF_RANGE,
+        };
+
+        let _reqid_msb = match value.get("reqid").and_then(|resource| resource.get("msb")) {
+            Some(_reqid_msb) => _reqid_msb
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u64>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _reqid_lsb = match value.get("reqid").and_then(|resource| resource.get("lsb")) {
+            Some(_reqid_lsb) => _reqid_lsb
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u64>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let ___reqid = UUID {
+            msb: _reqid_msb,
+            lsb: _reqid_lsb,
+            special_fields: SpecialFields::default(),
+        };
+      
+        let __reqid = MessageField(Some(Box::new(___reqid)));
+
+        let _token = match value.get("token") {
+            Some(_token) => _token
+                .as_str()
+                .unwrap_or_else(|| panic!("Deserialize: something wrong with token field")),
```

Comment:
Generally throughout the file -- are panics acceptable ways to communicate errors? Sometimes they are so I'm asking. More typically you'd load something into the unhappy path side of `Result<Self, D::Error>`, by doing e.g.
`Err(MyCoolError)`